### PR TITLE
creating initial contributing guideline

### PR DESCRIPTION
**Pull request by polatengin**
adding guideline for the people who wants to contribute to the BenchPress

### CODE REVIEW

File: docs/contributing_modules.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+├───{Resource Type Name}.Tests.ps1
+├───{Resource Type Name}.bicep
+└───{Resource Type Name}_Example.md
+```
+
+## Module Development
+
+- Create a new module with the required files mentioned above.
+
+- Add `Confirm-{Resource Type Name}` function to the [Modules/BenchPress.Azure/BenchPress.Azure.psd1](../Modules/BenchPress.Azure/BenchPress.Azure.psd1) file in the alphabetical order.
```
Comment (PabloZaiden) [inline docs/contributing_modules.md:26]:
```suggestion
- Add the `Confirm-{Resource Type Name}` function to the [Modules/BenchPress.Azure/BenchPress.Azure.psd1](../Modules/BenchPress.Azure/BenchPress.Azure.psd1) file in alphabetical order.
```

File: docs/contributing_modules.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+- Add `{Resource Type Name}` to the [Modules/BenchPress.Azure/Classes/ResourceType.psm1](../Modules/BenchPress.Azure/Classes/ResourceType.psm1) file in the alphabetical order.
```
Comment (PabloZaiden) [inline docs/contributing_modules.md:28]:
```suggestion
- Add `{Resource Type Name}` to the [Modules/BenchPress.Azure/Classes/ResourceType.psm1](../Modules/BenchPress.Azure/Classes/ResourceType.psm1) file in alphabetical order.
```

File: docs/contributing_modules.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+- Add `. $PSScriptRoot/Confirm-{Resource Type Name}.ps1` to the beginning of the [Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1](../Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1) file in the alphabetical order.
```
Comment (PabloZaiden) [inline docs/contributing_modules.md:30]:
```suggestion
- Add `. $PSScriptRoot/Confirm-{Resource Type Name}.ps1` to the beginning of the [Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1](../Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1) file in alphabetical order.
```

File: docs/contributing_modules.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+- Add a section for the new resource type to the end of the [Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1](../Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1) file in the alphabetical order, such as;
```
Comment (PabloZaiden) [inline docs/contributing_modules.md:32]:
```suggestion
- Add a section for the new resource type to the end of the [Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1](../Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1) file in alphabetical order, such as;
```

File: docs/contributing_modules.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+```powershell
+"{Resource Type Name}" {
+  return Confirm-{Resource Type Name} -ResourceGroupName $ResourceGroupName -Name $ResourceName
+}
+```
+
+## Module Testing
+
+- Run the following command in the root directory of the project to create local copy of the BenchPress in the `bin` folder;
```
Comment (PabloZaiden) [inline docs/contributing_modules.md:42]:
```suggestion
- Run the following command in the root directory of the project to create a local copy of BenchPress in the `bin` folder;
```

File: docs/contributing_modules.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+```powershell
+# old
+Import-Module Az.InfrastructureTesting
+
+# new
+Import-Module ../../bin/BenchPress.Azure.psd1
+```
+
+- Run this command to and create a new service principal and store the credentials in a variable;
```
Comment (PabloZaiden) [inline docs/contributing_modules.md:58]:
```suggestion
- Run this command to create a new service principal and store the credentials in a variable;
```

File: docs/contributing_modules.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+Import-Module Az.InfrastructureTesting
+```
+
+- Create a PR on the [BenchPress](https://github.com/azure/benchpress) repository. BenchPress Maintainers team will review the PR and merge it.
+
+## What if the resource type is not in the Azure PowerShell module?
+
+If the resource type is not in the Core Azure PowerShell module (`Az`), you have to do the following steps;
+
+- Find the module that contains the resource type. For example, `Az.Portal` module contains `Azure Dashboard` resource type.
```
Comment (PabloZaiden) [inline docs/contributing_modules.md:105]:
```suggestion
- Find the module that contains the resource type. For example, `Az.Portal` module contains the `Azure Dashboard` resource type.
```

**Review by PabloZaiden — CHANGES_REQUESTED** [review]
Minor changes

File: docs/contributing_modules.md (hunk 1 above)
Comment (polatengin) [reply docs/contributing_modules.md]: done! thanks 👍

File: docs/contributing_modules.md (hunk 2 above)
Comment (polatengin) [reply docs/contributing_modules.md]: done! thanks 👍

File: docs/contributing_modules.md (hunk 3 above)
Comment (polatengin) [reply docs/contributing_modules.md]: done! thanks 👍

File: docs/contributing_modules.md (hunk 4 above)
Comment (polatengin) [reply docs/contributing_modules.md]: done! thanks 👍

File: docs/contributing_modules.md (hunk 5 above)
Comment (polatengin) [reply docs/contributing_modules.md]: done! thanks 👍

File: docs/contributing_modules.md (hunk 6 above)
Comment (polatengin) [reply docs/contributing_modules.md]: moved this section to the getting_started.md file, thanks 👍

File: docs/contributing_modules.md (hunk 7 above)
Comment (polatengin) [reply docs/contributing_modules.md]: done! thanks 👍

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -76,26 +58,117 @@ the WSL2 subsystem with the dev container.
     1. Install .NET Core: `sudo apt-get update && sudo apt-get install -y dotnet-runtime-7.0`
     1. Restart the WSL terminal.
     1. Create a directory for source code and `cd` into it.
-    1. Clone or fork the repo as described below in
-       [Contributing to Code and Documentation](#contributing-to-code-and-documentation)
+    1. Clone or fork the repo as described in [Contributing to Code and Documentation](#contributing-to-code-and-documentation)
     1. `cd` into the repo's directory
-    1. Open VSCode from the command line with: `code .`
-    1. From the command window (Ctrl + Shift + P) choose `Reopen in Container`
+    1. Open VSCode from the command line with: `code .` (or `code-insiders .` if using the Insiders build)
```
Comment (hattan) [inline CONTRIBUTING.md:63]: Please link to a vs code insiders page that explains what the Insiders build is.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -76,26 +58,117 @@ the WSL2 subsystem with the dev container.
...
+
+examples/{Resource Type Name}
+├───{Resource Type Name}.Tests.ps1
+├───{Resource Type Name}.bicep
+└───{Resource Type Name}_Example.md
+```
+
+- Create a new module with the required files mentioned above.
+
+- Add `Confirm-{Resource Type Name}` function to the [Modules/BenchPress.Azure/BenchPress.Azure.psd1](../Modules/BenchPress.Azure/BenchPress.Azure.psd1) file in alphabetical order.
```
Comment (hattan) [inline CONTRIBUTING.md:100]:
```suggestion
- Add a `Confirm-{Resource Type Name}` function to the [Modules/BenchPress.Azure/BenchPress.Azure.psd1](../Modules/BenchPress.Azure/BenchPress.Azure.psd1) file in alphabetical order.
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -76,26 +58,117 @@ the WSL2 subsystem with the dev container.
...
+- Replace the `BeforeAll` section of the `examples/{Resource Type Name}/{Resource Type Name}.Tests.ps1` file with the following code;
+
+> Don't forget to replace it back before submitting a pull request.
+
+```powershell
+# old
+Import-Module Az.InfrastructureTesting
+
+# new
+Import-Module ../../bin/BenchPress.Azure.psd1
```
Comment (hattan) [inline CONTRIBUTING.md:131]: Would this work with just Import-Module BenchPress.Azure ? After running build.ps1 - Import

**Review by hattan — APPROVED** [review]
Left a few Nit comments, but looks good!

**Review by PabloZaiden — APPROVED** [review]

File: CONTRIBUTING.md (hunk 9 above)
Comment (polatengin) [reply CONTRIBUTING.md:100]: thanks 👍

File: CONTRIBUTING.md (hunk 8 above)
Comment (polatengin) [reply CONTRIBUTING.md:63]: done! 👍

File: CONTRIBUTING.md (hunk 10 above)
Comment (polatengin) [reply CONTRIBUTING.md]: yes, I updated the section, also added a note that has a link to build.ps1 script commands descriptions 👍

**Pull request merged**